sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CorrelationRuleFilter.java

```diff
@@ -285,8 +286,45 @@ public String toString() {
         return builder.toString();
     }
 
+    /**
+     *  Compares this RuleFilter to the specified object. The result is true if and only if the argument is not null
+     *  and is a CorrelationRuleFilter object that with the same parameters as this object.
+     *
+     * @param other - the object to which the current CorrelationRuleFilter should be compared.
+     * @return True, if the passed object is a CorrelationRuleFilter with the same parameter values, False otherwise.
+     */
+    @Override
+    public boolean equals(Object other) {
```

Comment:
You are missing the null check on `other`.

Reply:
instanceof operator should be covering the null check as well

Reply:
Good point!